platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/SecurityUtils.java

```diff
@@ -0,0 +1,166 @@
+        throw (E) e;
+      }
+    }
+  }
+
+  /**
+   * Performs the specified action with privileges enabled if the specified condition is <code>true
+   * </code>. The action is performed with <i>all</i> of the permissions possessed by this class'
+   * (or by the migration framework's) protection domain. Otherwise performs the specified action
+   * using the caller's permissions.
+   *
+   * <p>If the action's {@link ThrowingSupplier#get} method throws an <i>unchecked</i> exception, it
+   * will propagate through this method.
+   *
+   * <p><i>Note:</i> Any DomainCombiner associated with the current AccessControlContext will be
+   * ignored while the action is performed.
+   *
+   * @param <T> the type of the value returned by the action
+   * @param <E> the type of exceptions thrown by the action
+   * @param condition <code>true</code> to perform the action with the permissions possessed by this
+   *     class; <code>fase</code> to use the caller's permissions
+   * @param action the action to be performed
+   * @return the value returned by the action's {@code run} method
+   * @throws E if the specified action's threw the exception
+   * @throws IllegalArgumentException if <code>action</code> is <code>null</code>
+   */
+  public static <T, E extends Exception> T doConditionnalyPrivileged(
+      boolean condition, ThrowingSupplier<T, E> action) throws E {
+    if (condition) {
+      return SecurityUtils.doPrivileged(action);
+    }
+    Validate.notNull(action, SecurityUtils.INVALID_NULL_ACTION);
+    return action.get();
+  }
+
+  /**
+   * Performs the specified action with privileges enabled. The action is performed with <i>all</i>
+   * of the permissions possessed by this class' (or by the migration framework's) protection
+   * domain.
+   *
+   * <p>If the action's {@link ThrowingSupplier#get} method throws an <i>unchecked</i> exception, it
+   * will propagate through this method.
+   *
+   * <p><i>Note:</i> Any DomainCombiner associated with the current AccessControlContext will be
+   * ignored while the action is performed.
+   *
+   * @param <E> the type of exceptions thrown by the action
+   * @param action the action to be performed
+   * @throws E if the specified action's threw the exception
+   * @throws IllegalArgumentException if <code>action</code> is <code>null</code>
+   */
+  public static <E extends Exception> void doPrivileged(ThrowingRunnable<E> action) throws E {
+    Validate.notNull(action, SecurityUtils.INVALID_NULL_ACTION);
+    try {
+      AccessController.doPrivileged(
+          new PrivilegedExceptionAction<Void>() {
+            @Override
+            public Void run() throws Exception {
+              action.run();
+              return null;
+            }
+          });
+    } catch (PrivilegedActionException pe) {
+      final Exception e = pe.getException();
+
+      if (e instanceof RuntimeException) { // should never happen but just to be safe!
```

Comment:
✏️ Same comment about this not being worth cluttering code to check as it will not happen.

Reply:
Still prefer to be safe